File: Unique Number of Occurrences[13-July-2023]/Solution.py
#   >> PYTHON CODE <<

class Solution:
    def isFrequencyUnique(self, n : int, arr : List[int]) -> bool:
        mp={}
        
        for i in range(n):
            mp[arr[i]]=mp.get(arr[i],0)+1
        
        st=set()
        
        
        for key in mp.values():
            if(key in st): return False
            st.add(key)
        
        return True
        

File: Unique Number of Occurrences[13-July-2023]/Solution.java
//  >> JAVA CODE <<

class Solution {
    public static boolean isFrequencyUnique(int n, int[] arr) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            mp.put(arr[i], mp.getOrDefault(arr[i], 0) + 1);
        }
        
        HashSet<Integer> st = new HashSet<>();
        
        for (int value : mp.values()) {
            if (st.contains(value)) {
                return false;
            }
            st.add(value);
        }
        
        return true;
    }
}
    